src/components/molecules/OakModal/OakModal.tsx

```diff
@@ -0,0 +1,196 @@
+
+/**
+ * Modal dialog with trapped focus and a close button.
+ */
+export const OakModal = ({
+  children,
+  footerSlot,
+  domContainer = document.body,
+  isOpen,
+  onClose,
+  ...rest
+}: OakModalProps) => {
+  const [canaryElement, setCanaryElement] = useState<HTMLDivElement | null>(
+    null,
+  );
+  const [isScrolled, setIsScrolled] = useState(false);
+  const transitionRef = useRef<HTMLDivElement>(null);
+
+  useLayoutEffect(() => {
+    if (!canaryElement) {
+      return;
+    }
+    const observer = new IntersectionObserver(
+      (mutations) => {
+        setIsScrolled(!mutations.some((mutation) => mutation.isIntersecting));
+      },
+      {
+        root: canaryElement.parentElement,
+      },
+    );
+    observer.observe(canaryElement);
+
+    return () => {
+      observer.disconnect();
+    };
+  }, [canaryElement]);
+
+  return createPortal(
+    <Transition
+      in={isOpen}
+      nodeRef={transitionRef}
+      addEndListener={(done) => {
+        transitionRef.current?.addEventListener("transitionend", done);
+      }}
+      timeout={500}
+      mountOnEnter
+      unmountOnExit
+    >
+      {(state) => (
+        <FocusOn onEscapeKey={onClose} returnFocus autoFocus>
+          <FadeOutBox
+            $position="fixed"
+            $inset="all-spacing-0"
+            $zIndex="modal-dialog"
+            $background="black"
+            $opacity="semi-transparent"
+            $state={state}
+            $transition="standard-ease"
+          />
+          <SlideInFlex
+            ref={transitionRef}
+            $background="bg-primary"
+            $position="fixed"
+            $left="all-spacing-0"
+            $top="all-spacing-0"
+            $bottom="all-spacing-0"
+            $width="all-spacing-22"
+            $maxWidth="100vw"
+            $zIndex="modal-dialog"
+            $flexDirection="column"
+            $transition="standard-ease"
+            role="dialog"
+            $state={state}
+            {...rest}
+          >
+            <OakFlex
+              $ma="space-between-s"
+              $justifyContent="space-between"
+              $alignItems="center"
+            >
+              <OakImage
+                src={logoSrc}
+                $height="all-spacing-8"
+                $width="all-spacing-7"
+                alt=""
+              />
+              <InternalShadowRoundButton
+                onClick={onClose}
+                aria-label="Close"
+                defaultIconBackground="transparent"
+                defaultIconColor="black"
+                defaultTextColor="transparent"
+                hoverTextColor="transparent"
+                disabledTextColor="transparent"
+                hoverIconBackground="black"
+                hoverIconColor="white"
+                disabledIconBackground="transparent"
+                iconBackgroundSize="all-spacing-6"
+                iconSize="all-spacing-6"
+                iconName="cross"
+              />
```

Comment:
This button has a new component smell about it — but I wouldn't want to introduce a new flavour of button until there is another example of it in the wild. This might just be a one off "cross" to close the modal after all and the `Oak*` namespace is getting muddy already.